### DISCUSSION TITLE
How to handle not registered route in Grapevine?

### DISCUSSION THREAD

**Discussion by ThreeStars9908** [scottoffen/grapevine#149] [Q&A] [answered]
Hello, Scott!

I want to implement this feature using Grapevine.

When Grapevine raises an error because of unregistered route, I want it to be intercepted and return a standard JSON reply with a 200 status code, like this: `{ "error": { "message": "error message", "code": "error code" }}` instead of mirroring it to the browser.

Could you please help me with this?

Thanks!

**Comment by scottoffen** [comment, marked as answer]
If no route is found to handle the request (or if no route sends a response and one is required), then the status of the response is set to 501 Not Implemented. You can intercept this by providing either a global or local error handler.

> [!NOTE]
> Global error handlers are available on all `Router` instances, whereas local error handlers are specific to an instance. Unless your use case specifically calls for multiple servers, creating a local error handler is what you want to do.

```csharp
(server.Router as Router).LocalErrorHandlers.Add(HttpStatusCode.NotImplemented, async (ctx, ex) =>
{
    await ctx.Response.SendResponseAsync("Your response goes here");
});
```

**Reply by ThreeStars9908** [reply]
I couldn't understand.
Could you explain me in more details?
```
server.Router.Register<RestResources>();

server.Start();
Dispatcher.Invoke(() => MessageBox.Show($"Server started on port {PORT_NUM}.", "Success", MessageBoxButton.OK, MessageBoxImage.Information));
SetStatus($"READY! Server running on port {PORT_NUM}");
```
Thanks
